protocol/src/main/java/org/fido/iot/protocol/epid/EpidMaterialService.java

```diff
@@ -0,0 +1,118 @@
+// Copyright 2020 Intel Corporation
+// SPDX-License-Identifier: Apache 2.0
+
+package org.fido.iot.protocol.epid;
+
+import com.google.common.base.Joiner;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import javax.xml.bind.DatatypeConverter;
+import org.fido.iot.protocol.Composite;
+import org.fido.iot.protocol.Const;
+import org.fido.iot.protocol.InvalidMessageException;
+
+public class EpidMaterialService {
+
+  private final EpidUtils epidUtils = new EpidUtils();
+
+  /**
+   * Combines list of byte array into a single byte array.
+   *
+   * @param data List of byte array containing EPIDresource data
+   * @return byte array of all EPID resources
+   * @throws IOException for unhadled IO exceptions
+   */
+  public static byte[] createLVs(List<byte[]> data) throws IOException {
+
+    return (data.stream()
+        .collect(
+            () -> new ByteArrayOutputStream(),
+            (b, e) -> {
+              try {
+                b.write(e);
+              } catch (IOException ioException) {
+                throw new RuntimeException(ioException);
+              }
+            },
+            (a, b) -> {})
+        .toByteArray());
```

Comment:
Why not use the simpler

	return Stream.of(lists)
				.flatMap(x -> x.stream())
				.collect(Collectors.toList());

?

Reply:
This will not work here since function's return type is byte[] whereas the functional code in the comment will return List<byte[]>. Additionally even if we remove the collect statement the remaining functional code will not convert it to byte array.